src/app.rs

```diff
@@ -51,26 +51,22 @@ impl Application {
     pub fn prepare_from_opts(opts: Opts) -> Result<Self, exitcode::ExitCode> {
         openssl_probe::init_ssl_cert_env_vars();
 
-        let level = opts.log_level();
+        let level = std::env::var("LOG").unwrap_or_else(|_| match opts.log_level() {
+            "off" => "off".to_owned(),
+            level => [
+                format!("vector={}", level),
+                format!("codec={}", level),
+                format!("file_source={}", level),
+                "tower_limit=trace".to_owned(),
+                format!("rdkafka={}", level),
+            ]
+            .join(","),
+        });
+
```

Comment:
I think I miss something but isn't this code looks the same as it was? What is the difference?

Reply:
The difference is `level` and `levels`.

Reply:
Thanks. I opened the whole file and found that we log `level` later while applying`levels`. Now it's clear.